### PR TITLE
Graphql loader

### PR DESCRIPTION
**Pull request by BrandonHermanAvant**
Added webpack graphql loader which uses @gql2ts/from-query

### CODE REVIEW

File: packages/loader/__tests__/index-test.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000;
+  });
+
+  afterEach(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+  });
+
+  describe('Mutation', () => {
+    [
+      'mutationNoArgs',
```
Comment (BrandonHermanAvant) [inline packages/loader/__tests__/index-test.ts]: All of these tests may be overkill, but wanted to add some in to test different situations. Same with the 'Query' tests.

Comment (brettjurgens) [reply packages/loader/__tests__/index-test.ts]: @BrandonHermanAvant I agree that these are overkill. I think we should just test that the things are passed to `@gql2ts/from-query` which is already tested pretty well 😄

File: packages/loader/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  "peerDependencies": {
+    "graphql": ">= 0.10 <1.0.0"
+  },
+  "dependencies": {
+    "@gql2ts/from-query": "^1.8.1",
+    "@gql2ts/types": "^1.8.1",
+    "graphql-tag": "^2.9.2",
+    "loader-utils": "^1.1.0",
+    "prettier": "^1.14.0"
```
Comment (brettjurgens) [inline packages/loader/package.json]: let's change `prettier` to a peerDependency and do `>=1.4.0` (when TS support was added)

File: packages/loader/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// tslint:disable-next-line:typedef
+module.exports = function (source: any) {
+  const options: loaderUtils.OptionObject = loaderUtils.getOptions(this);
+
+  if (!options.schema) {
+    throw new Error('options[\'schema\'] must be set for graphql-loader');
+  }
+
+  fs.writeFileSync(`${this.resourcePath}.d.ts`, FileContent(options.schema, source, options.typeMap, options.options));
+  return `export default ${JSON.stringify(gql(source))};`;
```
Comment (brettjurgens) [inline packages/loader/src/index.ts]: i think we should just pass the input back and then encourage users to use `graphql-tag/loader`

File: packages/loader/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// tslint:disable:no-invalid-this
+// tslint:disable-next-line:typedef
+module.exports = function (source: any) {
+  const options: loaderUtils.OptionObject = loaderUtils.getOptions(this);
+
+  if (!options.schema) {
+    throw new Error('options[\'schema\'] must be set for graphql-loader');
+  }
+
+  fs.writeFileSync(`${this.resourcePath}.d.ts`, FileContent(options.schema, source, options.typeMap, options.options));
```
Comment (brettjurgens) [inline packages/loader/src/index.ts]: we can probably do this async

File: packages/loader/package.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "memory-fs": "^0.4.1",
+    "webpack": "^4.16.3"
+  },
+  "peerDependencies": {
+    "graphql": ">= 0.10 <1.0.0"
+  },
+  "dependencies": {
+    "@gql2ts/from-query": "^1.8.1",
+    "@gql2ts/types": "^1.8.1",
+    "graphql-tag": "^2.9.2",
```
Comment (brettjurgens) [inline packages/loader/package.json]: we should remove this

**Comment by brettjurgens** [comment]
@BrandonHermanAvant can you look into why coverage has decreased so much?

File: packages/loader/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+const DeclarationContent: string = `/* this is an auto-generated file. do not modify. */
+
+import { DocumentNode } from 'graphql';
+
+declare let GraphQLQuery: DocumentNode;
+
+export default GraphQLQuery;
+`;
+
+const FileContent: any = (
```
Comment (brettjurgens) [inline packages/loader/src/index.ts]: this shouldn't be any. and realistically it should be `typeof gqlRun`

File: packages/loader/src/__tests__/index-test.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { Stats } from 'webpack';
+import compiler from './__helpers__/compiler';
+import * as fromQuery from '@gql2ts/from-query';
+import buildDeclaration from '../buildDeclaration';
+import { PossibleSchemaInput } from '../../../types/node_modules/@gql2ts/util';
+import { ITypeMap, IProvidedOptions } from '../../../from-query/node_modules/@gql2ts/types';
```
Comment (brettjurgens) [inline packages/loader/src/__tests__/index-test.ts]: These two paths above need to be fixed

File: packages/loader/Readme.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+...
+  module: {
+    rules: [
+      {
+        test: /\.(graphql|gql)$/,
+        exclude: /node_modules/,
+        use: {
+          loader: "@gql2ts/loader",
+          options: {
+            schema: graphqlSchema, // Required
```
Comment (brettjurgens) [inline packages/loader/Readme.md]: Let's add a note somewhere about recommending `graphql-tag/loader`

File: packages/loader/src/__tests__/index-test.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { Stats } from 'webpack';
+import compiler from './__helpers__/compiler';
+import * as fromQuery from '@gql2ts/from-query';
```
Comment (brettjurgens) [inline packages/loader/src/__tests__/index-test.ts]: Let's put external imports above internal ones

File: packages/loader/src/index.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  const declaration: string = gqlRun(schema, source, typeMap, options)
+    .map(({ result }) => result)
+    .join('\n');
+
+  fs.writeFile(
+    `${this.resourcePath}.d.ts`,
+    buildDeclaration(declaration),
+    err => {
+      callback(err, source);
+    },
```
Comment (brettjurgens) [inline packages/loader/src/index.ts]: Let's get rid of the braces

File: packages/loader/src/__tests__/index-test.ts (hunk 7 above)
Comment (brettjurgens) [reply packages/loader/src/__tests__/index-test.ts]: (this is why tests are broken)

File: package.json [hunk 11]
```diff
@@ -38,20 +38,26 @@
   },
   "perferGlobal": true,
   "jest": {
+    "collectCoverageFrom": [
+      "**/src/**/*.ts",
+      "!cli",
```
Comment (brettjurgens) [inline package.json]: Let's do `!packages/cli` to be more specific

File: packages/loader/Readme.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+## Installation
+
+```shell
+npm install @gql2ts/loader
+```
+
+## Basic Usage in webpack configuration file:
+
+```javascript
+const graphqlSchema = fs.readFileSync('./config/schema.graphql').toString();
```
Comment (brettjurgens) [inline packages/loader/Readme.md]: I would just do `import Schema from './schema';`

File: packages/loader/Readme.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  module: {
+    rules: [
+      {
+        test: /\.(graphql|gql)$/,
+        exclude: /node_modules/,
+        use: {
+          loader: "@gql2ts/loader",
+          options: {
+            schema: graphqlSchema, // Required
+            typeMape: { }, // Optional: Partial<ITypeMap> from @gql2ts/types
```
Comment (brettjurgens) [inline packages/loader/Readme.md]: This is a typo

File: packages/loader/package.json [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    "build:watch": "tsc -w -p tsconfig.json"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/avantcredit/gql2ts.git"
+  },
+  "keywords": [
+    "gql2ts",
+    "graphql",
+    "typescript"
```
Comment (brettjurgens) [inline packages/loader/package.json]: Let's add Webpack

File: packages/loader/src/__tests__/index-test.ts (hunk 7 above)
Comment (BrandonHermanAvant) [reply packages/loader/src/__tests__/index-test.ts]: 🤦‍♂️ auto imported use these paths, should've caught this!

**Pull request merged**